buffer-tests/src/test/java/io/netty/buffer/api/tests/BufferTestSupport.java

```diff
@@ -323,88 +324,91 @@ static void stopExecutor() throws IOException {
-        assertThrows(IllegalStateException.class, () -> buf.setShort(0, (short) 32));
-        assertThrows(IllegalStateException.class, () -> buf.setUnsignedShort(0, 32));
-        assertThrows(IllegalStateException.class, () -> buf.setMedium(0, 32));
-        assertThrows(IllegalStateException.class, () -> buf.setUnsignedMedium(0, 32));
-        assertThrows(IllegalStateException.class, () -> buf.setInt(0, 32));
-        assertThrows(IllegalStateException.class, () -> buf.setUnsignedInt(0, 32));
-        assertThrows(IllegalStateException.class, () -> buf.setFloat(0, 3.2f));
-        assertThrows(IllegalStateException.class, () -> buf.setLong(0, 32));
-        assertThrows(IllegalStateException.class, () -> buf.setDouble(0, 32));
-
-        assertThrows(IllegalStateException.class, () -> buf.ensureWritable(1));
-        assertThrows(IllegalStateException.class, () -> buf.fill((byte) 0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.readByte());
+        assertThrows(UnsupportedOperationException.class, () -> buf.readUnsignedByte());
+        assertThrows(UnsupportedOperationException.class, () -> buf.readChar());
+        assertThrows(UnsupportedOperationException.class, () -> buf.readShort());
+        assertThrows(UnsupportedOperationException.class, () -> buf.readUnsignedShort());
+        assertThrows(UnsupportedOperationException.class, () -> buf.readMedium());
+        assertThrows(UnsupportedOperationException.class, () -> buf.readUnsignedMedium());
+        assertThrows(UnsupportedOperationException.class, () -> buf.readInt());
+        assertThrows(UnsupportedOperationException.class, () -> buf.readUnsignedInt());
+        assertThrows(UnsupportedOperationException.class, () -> buf.readFloat());
+        assertThrows(UnsupportedOperationException.class, () -> buf.readLong());
+        assertThrows(UnsupportedOperationException.class, () -> buf.readDouble());
+
+        assertThrows(UnsupportedOperationException.class, () -> buf.getByte(0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.getUnsignedByte(0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.getChar(0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.getShort(0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.getUnsignedShort(0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.getMedium(0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.getUnsignedMedium(0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.getInt(0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.getUnsignedInt(0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.getFloat(0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.getLong(0));
+        assertThrows(UnsupportedOperationException.class, () -> buf.getDouble(0));
```

Comment:
All of these also need to assert on `BufferClosedException` – pure read methods cannot throw `BufferReadOnlyException`.